### ISSUE TITLE
Retorno información desde rutas

### ISSUE THREAD

**Issue by santilaguna** [IIC2413/Syllabus-2019-2#185]
Hola!

Con mi grupo tenemos tres dudas respecto de la entrega. 

1) No entendemos bien cómo debemos retornar la información desde las rutas. No sabemos si mostrarlo en pantalla como texto o retornar el JSON. En este caso, cuando tenemos varios atributos que retornar (como mensajes recibidos, información de usuario y mensajes enviados) ¿los ordenamos como nosotros queramos?

2) Cuando se indica que debemos crear "Una ruta /messages/:id que al recibir el id de un correo en la URL, obtenga toda su información asociada.". El parámetro id ingresado desde la url debe ser consultado en la BDD como _id, uid o el correo del usuario?

3) Nos confunde un poco el esquema indicado en el enunciado, pues se indica que los documentos son de la forma:

 {
"id": "Am6FA24F",
"content": "Hola socio tanto tiempo, ¿c´omo est´as?. Te cuento que seguimos
operando a pesar de que aprobaron un recurso de protecci´on
en nuestra contra. Nadie ha venido a fiscalizar esta mina de
cobre haha. Espero que tengas la misma suerte con tu
termoel´ectrica en Antofagasta amigo m´ıo. Saludos!",
"metadata": {
"time": "2019-04-17 05:20:37",
"sender": "Franke",
"receiver": "Anheuser-Busch Inbev SA",
} ,
}

Sin embargo, al revisar las documentos de usuarios y mensajes son de las siguientes formas:

{'_id': ObjectId('5d194fe482caefc3b7ff2d57'),
 'contrase▒a': 0,
 'correo': 'qleiva@hotmail.com',
 'nacimiento': '1988-12-09',
 'nacionalidad': 'SWAZI',
 'nombre': 'Lic. Ana Luisa Delgado',
 'uid': 100}

{'_id': ObjectId('5d194f8082caefc3b7ff29b9'),
 'date': '2015-08-10',
 'lat': 35.333726,
 'long': 25.374343,
 'message': 'Buena! Viste la teleserie?',
 'mid': 689,
 'receptant': 130,
 'sender': 132}

Agradeceríamos mucho nos pudieran ayudar

**Comment by nivek0o0** [comment]
Hola,
1. Las rutas deben retornar obligatoriamente un JSON, porque después serán consumidas por su aplicación en PHP. Dicho de otra forma, a la aplicación en PHP no le sirve de nada el texto impreso en pantalla. Si quieres retornar cosas diversas, puedes retornar un diccionario con los pares `key: value` que necesites.

2. El **id** que se pedirá en la ruta corresponde al atributo `id` que nosotros le asignaremos a los correos. No al ObjectId de mongo ni ningún otro. En el ejemplo del enunciado sería `"id": "Am6FA24F"`

3. Esos documentos de usuarios y mensajes son del semestre pasado y **NO** se ocuparán en este proyecto. Para esta entrega no les daremos datos de correos, así que tendrán que crear en su BD algunos de ejemplo para probar sus rutas, siguiendo el esquema del enunciado.

**Comment by tomcan127** [comment]
Hola,
Entonces al crear un mensaje, debemos asignarle un id entro del json?
Seria de la forma:
    {"id": 1
        "content": "Hola",
        "metadata": {
            "receiver": "receiver",
            "sender": "sender",
            "time": "2019-12-3 11:59:20"
        }
    }

**Comment by nivek0o0** [comment]
Sí, la api debe generar ese ID único y guardarlo como un atributo del nuevo mensaje. Puede ser alfanumérico o entero. 

Y ojalá retornen ese ID en la respuesta del POST (si la creación fue exitosa) para que después podamos buscarlo y corroborar que se guardó en la BD
